drivers/ethernet/dsa_ksz8xxx.c

```diff
@@ -57,9 +57,15 @@ static void dsa_ksz8xxx_write_reg(const struct ksz8xxx_data *pdev,
 		.count = 1
 	};
 
+  /*
 	buf[0] = KSZ8XXX_SPI_CMD_WR | ((reg_addr >> 7) & 0x1F);
 	buf[1] = (reg_addr << 1) & 0xFE;
 	buf[2] = value;
+  */
```

Comment:
As I've said - The code which you commented out works on `ip_k66f` board. I'm able to configure the `KSZ8794` switch (setup MAC address table, get port status info, read SoC ID) - which means that the specified address is correct (at least for NXP's SPI driver).